File: stepik-java-api/src/main/java/org/stepik/api/actions/StepikSubmissionsAction.java
package org.stepik.api.actions;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.client.StepikApiClient;
import org.stepik.api.queries.submissions.StepikSubmissionsGetQuery;
import org.stepik.api.queries.submissions.StepikSubmissionsPostQuery;

/**
 * @author meanmail
 */
public class StepikSubmissionsAction extends StepikAbstractAction {
    public StepikSubmissionsAction(@NotNull StepikApiClient stepikApiClient) {
        super(stepikApiClient);
    }

    @NotNull
    public StepikSubmissionsGetQuery get() {
        return new StepikSubmissionsGetQuery(this);
    }

    @NotNull
    public StepikSubmissionsPostQuery post() {
        return new StepikSubmissionsPostQuery(this);
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/actions/StepikSubmissionsAction.kt
package org.stepik.api.actions

import org.stepik.api.client.StepikApiClient
import org.stepik.api.queries.submissions.StepikSubmissionsGetQuery
import org.stepik.api.queries.submissions.StepikSubmissionsPostQuery

class StepikSubmissionsAction(stepikApiClient: StepikApiClient) : StepikAbstractAction(stepikApiClient) {
    
    fun get(): StepikSubmissionsGetQuery {
        return StepikSubmissionsGetQuery(this)
    }
    
    fun post(): StepikSubmissionsPostQuery {
        return StepikSubmissionsPostQuery(this)
    }
}
